lib/internal/abort_controller.js

```diff
@@ -241,9 +252,11 @@ class AbortSignal extends EventTarget {
         abortSignal(resultSignal, signal.reason);
         return resultSignal;
       }
+      const signalWeakRef = new SafeWeakRef(signal);
+      finalizers.register(resultSignal, signalWeakRef);
```

Comment:
I think the logic here is wrong. Each `signal[kDependantSignals].add` should be paired with a `finalizers.register`, right?

So I would move this one inside the if below `signal[kDependantSignals].add` and also add `finalizers.register` in the loop in the else below for composite signals.

Reply:
Makes sense. Will fix it.